pkg/version/version.go

```diff
@@ -12,6 +12,12 @@ import (
 // DO NOT EDIT
 var Version = undefinedVersion
 
+// ProxyInitVersion is the pinned version of the proxy-init, from
+// https://github.com/linkerd/linkerd2-proxy-init
+// This has to be kept in sync with the constraint version for
+// github.com/linkerd/linkerd2-proxy-init in /Gopkg.toml
+var ProxyInitVersion = "v1.0.0"
```

Comment:
AFAICT, this is currently hard-coded, right? I wonder if we should update the [`docker build`](https://github.com/linkerd/linkerd2/blob/bd4c2788fa19f31b0fb2f32254ac6fd52c737565/bin/_docker.sh#L45-L49) command to inject this value at build time. Something like
```
docker build  -ldflags "-X version.ProxyInitVersion=${PROXY_INIT_VERSION}"
```

Reply:
Yeah, it's hard-coded. I see how it can be cleaner to parametrize this kind of things, but this one will hardly ever change. Where would we be setting `PROXY_INIT_VERSION`?

Reply:
Probably in `bin/docker-build` and maybe `.travis.yml`, with hard-coded default if the `PROXY_INIT_VERSION` environment variable is empty. Yeah, I agree that since this doesn't change often, what we have here is fine, unless others have different opinion on it.